astropy/time/formats.py

```diff
@@ -969,8 +969,16 @@ def parse_string(self, timestr, subfmts):
                 self._scale = scale
                 self._fits_scale = fits_scale
                 self._fits_realization = fits_realization
-            elif (scale != self.scale or fits_scale != self._fits_scale or
-                  fits_realization != self._fits_realization):
+            elif scale == self.scale:
```

Comment:
Actually, we can simplify perhaps further, since really the setting has to happen only once; the mistake I made originally was to think that in that case `self._scale` would be `None`, when the only thing guaranteed to be the case is that `_fits_scale` is not set. So:
```
if self._fits_scale is None:
    self._fits_scale = fits_scale
    self._fits_realization = fits_realization
    if self._scale is None:
        self._scale = scale
if scale != self.scale or fits_scale != self._fits_scale or ... # as before
    raise ...
```